src/App.tsx

```diff
@@ -1,27 +1,210 @@
-import React from 'react';
-import './App.scss';
-
-interface Props {
-  onClick: () => void;
-}
-
-export const Provider: React.FC<Props> = React.memo(
-  ({ onClick, children }) => (
-    <button
-      type="button"
-      onClick={onClick}
-    >
-      {children}
-    </button>
-  ),
-);
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import { FilterBy, ErrorMessage } from './types/Constants';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { RequestTodo } from './types/RequestTodo';
+import { client } from './utils/fetchClient';
+import { Header } from './Components/Header';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { ErrorModal } from './Components/ErrorModal';
+
+const USER_ID = 10552;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [querySearch, setQuerySearch] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterBy) {
+        case FilterBy.ACTIVE:
+          return !todo.completed;
+        case FilterBy.COMPLETED:
+          return todo.completed;
+        case FilterBy.ALL:
+          return true;
+        default:
+          return todo;
+      }
+    });
+  }, [filterBy, todos]);
+
+  const getTodosServer = async () => {
+    try {
+      const arrayTodos = await getTodos(USER_ID);
+
+      setTodos(arrayTodos);
+    } catch (error) {
+      setErrorMessage(ErrorMessage.DownloadError);
+    }
+  };
+
+  useEffect(() => {
+    getTodosServer();
+  }, []);
+
+  const addTodo = async () => {
+    const newTodo: RequestTodo = {
+      userId: USER_ID,
+      completed: false,
+      title: querySearch,
+    };
+
+    try {
+      if (!querySearch.trim()) {
+        return;
+      }
+
+      await createTodo(USER_ID, newTodo);
+
+      setTempTodo({
+        id: 0,
+        ...newTodo,
+      });
+
+      await getTodosServer();
+    } catch (error) {
+      setErrorMessage(ErrorMessage.NotAdd);
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const handleDeleteTodo = async (todoId: number) => {
+    try {
+      const filtTodos = todos.filter(todo => todo.id !== todoId);
+
+      setTodos(filtTodos);
```

Comment:
Optional. If you update state on the basis of the previous one, you shoud do it through function: 
`setState(prevState => prevState.filter(...))`
This will guarantee the state is always fresh.